std/algorithm/iteration.d

```diff
@@ -4980,6 +4980,142 @@ private auto sumKahan(Result, R)(Result result, R r)
         assert(repeat(1.0, n).sum == n);
 }
 
+/**
+Finds the mean (colloquially known as the average) of a range.
+
+For built-in numerical types, accurate Knuth & Welford mean calculation
+is used. The default result type is `real`, which is more accurate,
+but far slower than `double` on many architectures.
+
+For user-defined types, element by element summation is used. Additionally
+an extra parameter `seed` is needed in order to correctly seed the summation
+with the equivalent to `0`.
+
+The first overload of this function will return `T.init` if the range
+is empty. However, the second overload will return `seed` on empty ranges.
+
+This function is $(BIGOH r.length).
+
+Params:
+    r = An $(REF_ALTTEXT input range, isInputRange, std,range,primitives)
+    seed = For user defined types. Should be equivalent to `0`.
+
+Returns:
+    The mean of `r` when `r` is non-empty.
+*/
+T mean(T = real, R)(R r)
```

Comment:
We should move away from the use of `real` as a default and treat it as a special interest type. Please use double here.